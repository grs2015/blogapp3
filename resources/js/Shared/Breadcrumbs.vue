<template>
    <Link as="div" :href="link">
        <q-breadcrumbs class="text-orange">
            <template #separator>
                <q-icon size="1.5em" name="chevron_right" color="primary" />
            </template>
            <template v-for="item in data" :key="item.label">
                <q-breadcrumbs-el :label="$t(item.label)" :icon="item.icon" :href="item.route" @click="link = item.route" />
            </template>
        </q-breadcrumbs>
    </Link>
</template>

<script setup>
import { Link } from '@inertiajs/inertia-vue3';
import { ref } from 'vue';
import { usePage } from '@inertiajs/inertia-vue3';

const props = defineProps({ data: Array })
const link = ref('/admin/posts')

</script>
