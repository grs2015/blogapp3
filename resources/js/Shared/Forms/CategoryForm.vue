<script setup lang="ts">
import { categoryData } from '@/Interfaces/PaginatedData';


interface Props {
    data: {
        category?: categoryData | null
    }
}

const props = defineProps<Props>()

</script>

<template>
    FFF
</template>
