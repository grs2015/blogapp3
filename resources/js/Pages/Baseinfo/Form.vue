<script setup lang="ts">

import { Head } from '@inertiajs/inertia-vue3'
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'
import { settingsBreadcrumbs } from '@/breadcrumbsData.js'
import { ref } from 'vue'
import { breadcrumbsData, baseinfoData } from '@/Interfaces/PaginatedData'

import SettingsForm from '@/Shared/Forms/SettingsForm.vue'

interface Props {
    model: {
        baseinfo: baseinfoData
    }
}
const props = defineProps<Props>()

const breadcrumbs = ref<breadcrumbsData[]>(settingsBreadcrumbs)

</script>

<template>
    <Head title="Blog Settings" />
    <Breadcrumbs :data="breadcrumbs" />
    <SettingsForm :data="props.model" />
</template>


