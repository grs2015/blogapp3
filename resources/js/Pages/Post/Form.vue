<script setup lang="ts">

import { Head } from '@inertiajs/inertia-vue3'
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'
import { postCreateBreadcrumbs } from '@/breadcrumbsData.js'
import { ref } from 'vue'
import { tagData, categoryData, breadcrumbsData } from '@/Interfaces/PaginatedData'
import PostCreateTable from '@/Shared/Tables/PostCreateTable.vue'

interface Props {
    model: {
        categories: Array<categoryData>,
        tags: Array<tagData>,
    }
}
const props = defineProps<Props>()

const breadcrumbs = ref<breadcrumbsData[]>(postCreateBreadcrumbs)

</script>


<template>
    <Head title="Blog Post - Create" />
    <Breadcrumbs :data="breadcrumbs" />
    <PostCreateTable :data="props.model" />
    <!-- {{ props.model.tags }} -->
</template>
