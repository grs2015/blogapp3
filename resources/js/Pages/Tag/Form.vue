<script setup lang="ts">

import { Head } from '@inertiajs/inertia-vue3'
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'
import { tagCreateBreadcrumbs, tagEditBreadcrumbs } from '@/breadcrumbsData.js'
import { ref } from 'vue'
import { tagData, breadcrumbsData } from '@/Interfaces/PaginatedData'
import TagForm from '@/Shared/Forms/TagForm.vue'

interface Props {
    model: {
        tag?: tagData | null
    }
}

const props = defineProps<Props>()

const breadcrumbs = ref<breadcrumbsData[]>(!props.model.tag ? tagCreateBreadcrumbs : tagEditBreadcrumbs)

const title = props.model.tag ? 'Edit' : 'Create'

</script>

<template>
    <Head :title="`Blog Tag - ${ title }`" />
    <Breadcrumbs :data="breadcrumbs" />
    <TagForm :data="props.model" />
</template>


