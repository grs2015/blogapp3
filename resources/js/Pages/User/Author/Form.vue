<script lang="ts">

import Layout from '@/Shared/LayoutAuthor.vue'

export default {
    layout: Layout,
}

</script>

<script setup lang="ts">

import { Head } from '@inertiajs/inertia-vue3'
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'
import { accountAuthorBreadcrumbs } from '@/breadcrumbsData.js'
import { ref } from 'vue'
import { breadcrumbsData, userData } from '@/Interfaces/PaginatedData'
import AccountForm from '@/Shared/Forms/AccountForm.vue'

interface Props {
    model: {
        user?: userData | null
    }
}
const props = defineProps<Props>()

const breadcrumbs = ref<breadcrumbsData[]>(accountAuthorBreadcrumbs)

</script>

<template>
    <Head title="User Account" />
    <Breadcrumbs :data="breadcrumbs" />
    <AccountForm :data="props.model" />
</template>
