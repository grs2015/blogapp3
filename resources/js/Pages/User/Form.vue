<script setup lang="ts">

import { Head } from '@inertiajs/inertia-vue3'
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'
import { accountBreadcrumbs } from '@/breadcrumbsData.js'
import { ref } from 'vue'
import { breadcrumbsData, userData } from '@/Interfaces/PaginatedData'
import AccountForm from '@/Shared/Forms/AccountForm.vue'

interface Props {
    model: {
        user?: userData | null
    }
}
const props = defineProps<Props>()

const breadcrumbs = ref<breadcrumbsData[]>(accountBreadcrumbs)

</script>

<template>
    <Head title="User Account" />
    <Breadcrumbs :data="breadcrumbs" />
    <AccountForm :data="props.model" />
</template>
