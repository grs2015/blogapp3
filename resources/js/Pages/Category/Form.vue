<script setup lang="ts">

import { Head } from '@inertiajs/inertia-vue3'
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'
import { categoryCreateBreadcrumbs, categoryEditBreadcrumbs } from '@/breadcrumbsData.js'
import { ref } from 'vue'
import { categoryData, breadcrumbsData } from '@/Interfaces/PaginatedData'
import CategoryForm from '@/Shared/Forms/CategoryForm.vue'

interface Props {
    model: {
        category?: categoryData | null
    }
}
const props = defineProps<Props>()

const breadcrumbs = ref<breadcrumbsData[]>(!props.model.category ? categoryCreateBreadcrumbs : categoryEditBreadcrumbs)

const title = props.model.category ? 'Edit' : 'Create'

</script>


<template>
    <Head :title="`Blog Category - ${ title }`" />
    <Breadcrumbs :data="breadcrumbs" />
    <CategoryForm :data="props.model" />
</template>
