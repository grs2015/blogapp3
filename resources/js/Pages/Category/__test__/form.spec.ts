import { mount } from "@vue/test-utils";
import Form from '@/Pages/Category/Form.vue'
import CategoryForm from '@/Shared/Forms/CategoryForm.vue'
import { Head } from "@inertiajs/inertia-vue3";
import Breadcrumbs from '@/Shared/Breadcrumbs.vue'

let wrapper = null;

interface Data {
    content?: string,
    icon?: string,
    id: number,
    color: string,
    'meta_title'?: string,
    'parent_id'?: number,
    slug?: string,
    title: string
}

const data: Data = {
    content: 'test category',
    icon: 'description',
    color: 'purple',
    id: 1,
    'meta_title': 'test category meta',
    slug: 'test-category',
    title: 'category title'
}

afterEach(async () => await wrapper.unmount())

describe('Category/Form component', () => {
    it("makes a snapshot", async () => {
        wrapper = await mount(Form, { props: { model: { category: data } }, shallow: true } )

        expect(wrapper.element).toMatchSnapshot();
    });

    it('checks the data from BE is defined on component if passed', async () => {

        wrapper = await mount(Form, { props: { model: { category: data } }, shallow: true } )

        expect(wrapper.props().model.category).toEqual({
            content: 'test category',
            icon: 'description',
            color: 'purple',
            id: 1,
            'meta_title': 'test category meta',
            slug: 'test-category',
            title: 'category title'
        })
    })

    it("checks the existence & renders the Head with appropriate title data when props passed", async () => {

        wrapper = await mount(Form, { props: { model: { category: data } }, shallow: true } )

        expect(wrapper.findComponent(Head).exists()).toBe(true);
        expect(wrapper.findComponent(Head).attributes("title")).toBe(
            "Blog Category - Edit"
        );
    });

    it("checks the existence & renders the Head with appropriate title data when no props passed", async () => {

        wrapper = await mount(Form, { props: { model: { category: null } }, shallow: true } )

        expect(wrapper.findComponent(Head).exists()).toBe(true);
        expect(wrapper.findComponent(Head).attributes("title")).toBe(
            "Blog Category - Create"
        );
    });

    it("checks the existence & passes to Breadcrumbs component appropriate data when props passed", async () => {

        wrapper = await mount(Form, { props: { model: { category: data } }, shallow: true } )

        const breadcrumbsComponent = wrapper.findComponent(Breadcrumbs);

        expect(breadcrumbsComponent.exists()).toBe(true);
        expect(breadcrumbsComponent.props().data).toEqual([
            { label: "Dashboard", icon: "home", route: "/admin" },
            { label: "Categories", icon: "category", route: "/admin/categories" },
            { label: "Edit", icon: "edit" }
        ]);
    });

    it("checks the existence & passes to Breadcrumbs component appropriate data when no props passed", async () => {

        wrapper = await mount(Form, { props: { model: { category: null } }, shallow: true } )

        const breadcrumbsComponent = wrapper.findComponent(Breadcrumbs);

        expect(breadcrumbsComponent.exists()).toBe(true);
        expect(breadcrumbsComponent.props().data).toEqual([
            { label: "Dashboard", icon: "home", route: "/admin" },
            { label: "Categories", icon: "category", route: "/admin/categories" },
            { label: "Create", icon: "edit" }
        ]);
    });

    it("checks the existence & passes to CategoryForm component the props", async () => {

        wrapper = await mount(Form, { props: { model: { category: data } }, shallow: true } )

        const categoryFormComponent = wrapper.findComponent(CategoryForm);

        expect(categoryFormComponent.exists()).toBe(true);
        expect(categoryFormComponent.props().data.category).toEqual(data)
    });
})
