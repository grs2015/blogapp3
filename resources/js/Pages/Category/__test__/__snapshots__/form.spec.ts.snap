// Vitest Snapshot v1

exports[`Category/Form component > makes a snapshot 1`] = `
<div
  data-v-app=""
>
  
  <head-stub
    title="Blog Category - Edit"
  />
  <breadcrumbs-stub
    data="[object Object],[object Object],[object Object]"
  />
  <category-form-stub
    data="[object Object]"
  />
  
</div>
`;
